sparta/sparta/utils/SpartaSharedPointer.hpp

```diff
@@ -122,6 +122,22 @@ namespace sparta
             ++ref_count_->count;
         }
 
+        /**
+         * \brief Take ownership of a reference pointer
+         * \param orig The original pointer
+         *
+         * This SpartaSharedPointer now replaces orig.  Orig becomes a
+         * nullptr with no references
+         */
+        SpartaSharedPointer(SpartaSharedPointer && orig) :
```

Comment:
I am a bit unsure about this. Typically, move constructors and move assigns should be ```nothrow``` methods. 
Can we just make the ```ref_count_``` of the moved from pointer point to ```nullptr``` instead of allocating a new ```RefCount```?
```
orig.ref_count_ = nullptr;
```

Reply:
Originally thought that, but then I have to add an `if` condition on all uses of `ref_count_` to make sure it isn't nullptr.  So, instead, I make the original look like it was constructed with `nullptr_t`.

Reply:
I'll investigate the `nothrow`.  Good catch.

Reply:
Done.  Although I couldn't add it to all operators as some do a check to ensure the user isn't doing something dumb (and that code throws an exception).